src/nodes.coffee

```diff
@@ -57,6 +60,34 @@ exports.Base = class Base
   compile: (o, lvl) ->
     fragmentsToText @compileToFragments o, lvl
 
+  # Occasionally a node is compiled multiple times, for example to get the name
+  # of a variable to add to scope tracking. When we know that a “premature”
+  # compilation won’t result in comments being output, set those comments aside
+  # so that they’re preserved for a later `compile` call that will result in
+  # the comments being included in the output.
+  compileWithoutComments: (o, lvl, method = 'compile') ->
+    if @comments
+      @ignoreTheseCommentsTemporarily = @comments
```

Comment:
This is a bit unpleasant. Could we get away with a flag such as `@ignoreComments = yes or no`, or would that result in peppering the code with `if @ignoreComments …`?

Alternatively, do they have to be assigned to `@` since they're always reassigned at the end of the method? I'd probably be less bothered if it was just a local variable.